api/transport_security.proto

```diff
@@ -0,0 +1,151 @@
+  // the Subject Public Key Information (SPKI) of the presented certificate.
+  // This is the same format as used in HTTP Public Key Pinning.
+  repeated string verify_spki_sha256 = 3;
+
+  // An optional list of subject alt names. If specified, Envoy will verify that
+  // the certificate’s subject alt name matches one of the specified values.
+  repeated string verify_subject_alt_name = 4;
+
+  // Must present a signed time-stamped OCSP response.
+  google.protobuf.BoolValue require_ocsp_staple = 5;
+
+  // Must present signed certificate time-stamp.
+  google.protobuf.BoolValue require_signed_certificate_timestamp = 6;
+}
+
+// TLS context shared by both client and server TLS contexts.
+message CommonTlsContext {
+  // TLS protocol versions, cipher suites etc.
+  TlsParameters tls_params = 1;
+
+  // Multiple TLS certificates can be associated with the same context,
+  // e.g. to allow both RSA and ECDSA certificates [V2-API-DIFF].
+  // TLS certificates can be either configured locally or fetched from SDS.
+  repeated TlsCertificate tls_certificates = 2;
+  repeated SdsSecretConfig tls_certificate_sds_secret_configs = 6;
+
+  // How to validate peer certificates.
+  CertificateValidationContext validation_context = 3;
+
+  // Protocols to negotiate over ALPN
+  repeated string alpn_protocols = 4;
+
+  // These fields are deprecated and only are used during the interim v1 -> v2
+  // transition period for internal purposes. They should not be used outside of
+  // the Envoy binary.
+  message DeprecatedV1 {
+    string alt_alpn_protocols = 1;
+  }
+  DeprecatedV1 deprecated_v1 = 5;
+}
+
+message UpstreamTlsContext {
+  CommonTlsContext common_tls_context = 1;
+
+  // SNI string to use when creating TLS backend connections.
+  string sni = 2;
+}
+
+// [V2-API-DIFF] This has been reworked to support alternative modes of
+// certificate/key delivery, for consistency with the upstream TLS context and
+// to segregate the client/server aspects of the TLS context.
+message DownstreamTlsContext {
+  CommonTlsContext common_tls_context = 1;
+
+  // If specified, Envoy will reject connections without a valid client
+  // certificate.
+  google.protobuf.BoolValue require_client_certificate = 2;
+
+  // If specified, Envoy will reject connections without a valid and matching SNI.
+  google.protobuf.BoolValue require_sni = 3;
+
+  oneof session_ticket_keys_type {
+    TlsSessionTicketKeys session_ticket_keys = 4;
+    SdsSecretConfig session_ticket_keys_sds_secret_config = 5;
+  }
+}
+
+message SdsSecretConfig {
+  // Name (FQDN, UUID, SPKI, SHA256, etc.) by which the secret can be uniquely referred to.
+  // When both name and config are specified, then secret can be fetched and/or reloaded via SDS.
+  // When only name is specified, then secret will be loaded from static resources [V2-API-DIFF].
+  string name = 1;
+  ConfigSource sds_config = 2;
+}
+
+message TransportSecurityContext {
```

Comment:
Nit: feel free to add this before `SdsSecretConfig`, so it's next to other context definition.

Reply:
Done